### PR TITLE
Adding a feature Metadata/ocean-basin-flag to all ioda converters

### LINKED ISSUE #775: Add the ocean basin flag to all converters
See [PR#732](https://github.com/NOAA-EMC/GDASApp/pull/732).

### PR DESCRIPTION
**Pull request by apchoiCMD**
- Moved a simple OceanMask struct that reads in regional ocean masks to base class
- Those are an implementation of the feature in all converters

Close https://github.com/NOAA-EMC/GDASApp/issues/775

- Tested with one full obs file for each converter:
ocean_flag_combined

### CODE REVIEW

File: utils/test/testinput/gdas_icecamsr2ioda.yaml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-
-test:
-  reference filename: testref/icecamsr2ioda.test
-  test output filename: testoutput/icecamsr2ioda.test
-  float relative tolerance: 1e-6
+provider: AMSR2
+window begin: 2018-04-15T06:00:00Z
+window end: 2018-04-15T12:00:00Z
+output file: icec_amsr2_north.ioda.nc
+#ocean basin: RECCAP2_region_masks_all_v20221025.nc
```
Comment (ShastriPaturi) [inline utils/test/testinput/gdas_icecamsr2ioda.yaml:5]: Is there a purpose to have the commented line here?

**Review by ShastriPaturi — CHANGES_REQUESTED** [review]

File: utils/test/testinput/gdas_icecamsr2ioda.yaml (hunk 1 above)
Comment (apchoiCMD) [reply utils/test/testinput/gdas_icecamsr2ioda.yaml:5]: > Is there a purpose to have the commented line here?

Like a previous [PR#732](https://github.com/NOAA-EMC/GDASApp/pull/732), I commented out ocean basin as default.

Comment (guillaumevernieres) [reply utils/test/testinput/gdas_icecamsr2ioda.yaml:5]: Yes @ShastriPaturi , it's a good reminder that it's not part of the ctest, and needs to be used when processing obs.

**Review by guillaumevernieres — APPROVED** [review]
Thanks @apchoiCMD , looks good.

**Review by ShastriPaturi — APPROVED** [review]
Thanks @apchoiCMD for putting this effort.
Much appreciated.

**Comment by apchoiCMD** [comment]
> Thanks @apchoiCMD for putting this effort. Much appreciated.

Thank you to both @guillaumevernieres  and @ShastriPaturi for your patience! I couldn't make it without your support.

**Pull request merged**